irc/server.py

```diff
@@ -436,23 +436,16 @@ def handle_dump(self, params):
                 print("     ", client.nick, client)
 
     def handle_ison(self, params):
+        response = '303: '
+        if len(params) == 0 or params.isspace():
+            response = '461: ISON Not enough parameters'
+            return response    
+        nickOnline = []
         for nick in params.split(" "):
-            if len(params) == 0 or params.isspace():
-                response = '461: ISON Not enough parameters'
-                return response
-
-            first = True
-            response = '303: '
-
-            for c in self.server.clients:
-                if c == nick:
-                    if not first:
-                        response += ' ' + nick
-                    else:
-                        response += nick
-                        first = False
-        return self.send_queue.append(response)
-
+            if nick in self.server.clients:
```

Comment:
@Querela mentioned that this would only work if self.server.clients happens to be a string. Although this is the same logic used for a user to use a nick in handle_nick.

Reply:
Hi @edman80, the new code looks good.
I only read through your code yesterday and hadn't taken more time to look through the rest of the file.
`if nick in self.server.clients:` could possibly have failed if clients were not really a string but a more complex object that may or may not have overloaded some magic (e.g. "__op__") methods. I couldn't test the code, so I mentioned my concerns (even if unfounded). :-)

Reply:
Understood, I'm not sure of a different method right now but I can take a look.

Reply:
Well ...
I think you should prefix the number response code with something like this:
`self.server.servername` (see lines server.py:408, server.py:315; line :317 for a number code; I'm not sure if the response should include the `servername` or the `client_ident()` - that's in the docs):

```
response = ':%s 303: ' % self.server.servername
# ...
    response = ':%s 461: ISON Not enough parameters' % self.server.servername
# ...
# or (without the first line):
response = ':%s 303: %s' % (self.server.servername, ' '.join(nickOnline))
return response
```

Reply:
I believe it should be prefixed with the servername and suffixed by the client_ident().nick